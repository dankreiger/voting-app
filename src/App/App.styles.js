import styled from 'styled-components';
import { media } from 'utils/style/style-utils';

export const AppContainer = styled.div`
  min-height: 80vh;
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 20px;
  ${media.mdQuery` 
    padding: 40px;`}
`;
