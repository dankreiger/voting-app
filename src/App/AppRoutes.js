import React from 'react';
import { Route, Switch } from 'react-router-dom';
import QuestionsList from 'components/QuestionsList/QuestionsList';
import QuestionsDetail from 'components/QuestionsDetail/QuestionsDetail';

const AppRoutes = () => {
  return (
    <Switch>
      <Route exact path="/" component={QuestionsList} />
      <Route exact path="/questions/:questionId" component={QuestionsDetail} />
    </Switch>
  );
};

export default AppRoutes;
