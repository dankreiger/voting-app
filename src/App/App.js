import React, { useEffect } from 'react';
import { arrayOf, bool, func } from 'prop-types';
import { connect } from 'react-redux';
import * as actions from 'actions';
import { AppContainer } from './App.styles';
import AppRoutes from './AppRoutes';
import { Question } from 'typings/Question.proptypes';
import LoadingSpinner from 'components/LoadingSpinner/LoadingSpinner';

function App({ fetchQuestions, loading, questions }) {
  useEffect(() => {
    fetchQuestions();
  }, [fetchQuestions]);

  const questionsAreLoaded = () => !loading && questions.length > 0;

  const renderApp = () => {
    if (loading) {
      return <LoadingSpinner />;
    } else if (questionsAreLoaded) {
      return <AppRoutes />;
    } else {
      return <h1>No questions available</h1>;
    }
  };
  return <AppContainer>{renderApp()}</AppContainer>;
}

App.propTypes = {
  fetchQuestions: func,
  loading: bool,
  questions: arrayOf(Question)
};

const mapStateToProps = ({ questionsReducer }) => ({
  loading: questionsReducer.loading,
  questions: questionsReducer.questions
});

export default connect(
  mapStateToProps,
  actions
)(App);
