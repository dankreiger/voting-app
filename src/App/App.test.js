import React from 'react';
import App from './App';

jest.mock('utils/test-services/fetchQuestions');

describe('App', () => {
  let component;
  const fetchQuestions = jest.fn();

  beforeEach(() => {
    component = global.setupReduxConnectedReactRouterComponent(
      <App fetchQuestions={fetchQuestions} />,
      'mount'
    );
  });

  describe('rendering', () => {
    test('renders as expected', () => {
      expect(component.length).toBe(1);
      expect(component).toMatchSnapshot();
    });
  });

  describe('fetching from api on mount', () => {
    test('calls fetchQuestions on mount', () => {
      requestAnimationFrame(done => {
        expect(fetchQuestions).toHaveBeenCalled();
        done();
      });
    });
  });
});
