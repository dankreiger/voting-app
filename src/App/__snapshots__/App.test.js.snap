// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App rendering renders as expected 1`] = `ReactWrapper {}`;
